src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedByKey: GroupsMap<T> = {};
+  const allKeys = items
+    .map((item) => item[key])
+    .filter((item, index, arr) => arr.indexOf(item) === index);
+
+  for (const currentKey of allKeys) {
+    groupedByKey[String(currentKey)] = items.filter(
+      (item) => item[key] === currentKey,
+    );
```

Comment:
The filtering operation inside the loop is not efficient as it iterates over the entire 'items' array for each unique key. Consider using a single iteration to build the groups, which will be more performant.